File: php/src/DigifactException.php
<?php

declare(strict_types=1);

namespace Digifact\Fel;

/**
 * Base exception for all Digifact FEL SDK errors.
 */
class DigifactException extends \RuntimeException
{
    protected ?int $apiCode;
    protected array $raw;

    public function __construct(string $message, ?int $apiCode = null, array $raw = [], \Throwable $previous = null)
    {
        parent::__construct($message, 0, $previous);
        $this->apiCode = $apiCode;
        $this->raw = $raw;
    }

    public function getApiCode(): ?int
    {
        return $this->apiCode;
    }

    public function getRaw(): array
    {
        return $this->raw;
    }
}

File: dotnet/Exceptions/DigifactException.cs
namespace Digifact.Fel;

/// <summary>Base exception for all Digifact SDK errors.</summary>
public class DigifactException : Exception
{
    /// <summary>API error code returned by Digifact, if available.</summary>
    public int? ApiCode { get; }

    /// <summary>Raw response body that triggered the exception, if available.</summary>
    public string? Raw { get; }

    /// <summary>Create a new <see cref="DigifactException"/>.</summary>
    /// <param name="message">Human-readable error message.</param>
    /// <param name="apiCode">Optional Digifact API error code.</param>
    /// <param name="raw">Optional raw response body for debugging.</param>
    public DigifactException(string message, int? apiCode = null, string? raw = null)
        : base(message)
    {
        ApiCode = apiCode;
        Raw = raw;
    }
}
